tempto-core/src/main/java/com/teradata/tempto/s3/S3Client.java

```diff
@@ -0,0 +1,232 @@
+ */
+
+import java.io.*;
+import java.util.List;
+import java.util.Comparator;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.output.ByteArrayOutputStream;
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectRequest;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.amazonaws.services.s3.model.ListObjectsV2Request;
+import com.amazonaws.services.s3.model.ListObjectsV2Result;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Ordering;
+import com.google.inject.name.Named;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * S3 client based on the Amazon AWS client API
+ */
+public class S3Client
+{
+    private static final Logger logger = getLogger(S3Client.class);
+    private static final String PATH_SEPARATOR = "/";
+    private static final Splitter PATH_SPLITTER = Splitter.on('/').omitEmptyStrings();
+
+    public static final String CONF_S3_AUTH_KEY_ID = "s3.auth.keyid";
+    public static final String CONF_S3_AUTH_SECRET = "s3.auth.secret";
+    public static final String CONF_S3_BUCKET_NAME = "s3.bucketname";
+
+    private final String bucketName;
+    private final AmazonS3 client;
+
+    public S3Client(
+            @Named(CONF_S3_AUTH_KEY_ID) String keyId,
+            @Named(CONF_S3_AUTH_SECRET) String secret,
+            @Named(CONF_S3_BUCKET_NAME) String bucketName)
+    {
+        // Check null values and set object variables
+        checkNotNull(keyId, "aws api access key id is null");
+        checkNotNull(secret, "aws api secret access key is null");
+        this.bucketName = checkNotNull(bucketName, "s3 bucket name is null");
+
+        // Create S3 client
+        client = AmazonS3ClientBuilder.standard()
+                .withCredentials(new AWSStaticCredentialsProvider(buildAWSCredentials(keyId, secret)))
+                .build();
+    }
+
+    private BasicAWSCredentials buildAWSCredentials(String keyId, String secret)
+    {
+        try {
+            return new BasicAWSCredentials(keyId, secret);
+        } catch (Exception e) {
+            throw new AmazonClientException("Cannot initialize credentials", e);
+        }
+    }
+
+    public boolean exists(String path)
+    {
+        return client.doesObjectExist(bucketName, path);
+    }
+
+    public List<String> listFiles(String path)
+    {
+        // Get list of all files located in this path
```

Comment:
I will change the comment to get list of all objects located in this path for both because that is what it is doing